### PR TITLE
ci: let Renovate check Java files for tags

### LINKED ISSUE #54: Configure Renovate to look for tags in Java code, and add such a tag to Versions.java

### PR DESCRIPTION
**Pull request by sverrehu**
Closes #54

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -3,4 +3,16 @@
...
+  "regexManagers": [
+    {
+      "fileMatch": [
+        ".+\\.java",
+      ],
+      "matchStrings": [
+        "renovate:\\s*datasource=(?<datasource>\\S+)\\s+depName=(?<depName>\\S+)(?:\\s+(package|lookup)Name=(?<packageName>\\S+))?(?:\\s+versioning=(?<versioning>\\S+))?(?:\\s+registryUrl=(?<registryUrl>\\S+))?\\n.+?_(?:VERSION|version|TAG|tag)\\s*[?=:]+\\s*\"?(?<currentValue>[^\" \\t\\n]+)\"?"
+      ],
+      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
+    }
```
Comment (erikgb) [inline .github/renovate.json5:16]:
```suggestion
    },
```

File: .github/renovate.json5 [hunk 2]
```diff
@@ -3,4 +3,16 @@
...
     "config:base",
   ],
+  "ignorePaths": ["**/node_modules/**", "**/bower_components/**"],
+  "regexManagers": [
+    {
+      "fileMatch": [
+        ".+\\.java",
+      ],
+      "matchStrings": [
+        "renovate:\\s*datasource=(?<datasource>\\S+)\\s+depName=(?<depName>\\S+)(?:\\s+(package|lookup)Name=(?<packageName>\\S+))?(?:\\s+versioning=(?<versioning>\\S+))?(?:\\s+registryUrl=(?<registryUrl>\\S+))?\\n.+?_(?:VERSION|version|TAG|tag)\\s*[?=:]+\\s*\"?(?<currentValue>[^\" \\t\\n]+)\"?"
```
Comment (erikgb) [inline .github/renovate.json5:13]:
```suggestion
        "renovate:\\s*datasource=(?<datasource>\\S+)\\s+depName=(?<depName>\\S+)(?:\\s+(package|lookup)Name=(?<packageName>\\S+))?(?:\\s+versioning=(?<versioning>\\S+))?(?:\\s+registryUrl=(?<registryUrl>\\S+))?\\n.+?_(?:VERSION|version|TAG|tag)\\s*[?=:]+\\s*\"?(?<currentValue>[^\" \\t\\n]+)\"?",
```

File: .github/renovate.json5 [hunk 3]
```diff
@@ -3,4 +3,16 @@
...
+      ],
+      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
```
Comment (erikgb) [inline .github/renovate.json5:15]:
```suggestion
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}",
```

**Review by erikgb — COMMENTED** [review]
Ugh, I don't like this advanced configuration of Renovate. But I will look another way. 😉 

Added some JSON5 nits.

**Review by erikgb — APPROVED** [review]

**Comment by sverrehu** [comment]
> Ugh, I don't like this advanced configuration of Renovate. But I will look another way. 😉

I don't know of a simpler way to accieve the same goal. Sorry.

**Pull request merged**